src/Castle.Core/DynamicProxy/ProxyGenerationOptions.cs

```diff
@@ -123,19 +125,69 @@ public MixinData MixinData
 			}
 		}
 
+		/// <summary>
+		///   Adds a delegate type to the list of mixins that will be added to generated proxies.
+		///   That is, generated proxies will have a `Invoke` method with a signature matching that
+		///   of the specified <paramref name="delegateType"/>.
+		/// </summary>
+		/// <param name="delegateType">The delegate type whose `Invoke` method should be reproduced in generated proxies.</param>
+		/// <exception cref="ArgumentNullException"><paramref name="delegateType"/> is <see langword="null"/>.</exception>
+		/// <exception cref="ArgumentException"><paramref name="delegateType"/> is not a delegate type.</exception>
+		public void AddDelegateTypeMixin(Type delegateType)
+		{
+			if (delegateType == null)
+			{
+				throw new ArgumentNullException(nameof(delegateType));
+			}
+
+			if (delegateType.GetTypeInfo().IsSubclassOf(typeof(MulticastDelegate)) == false)
+			{
+				throw new ArgumentException("Type must be a delegate type.", nameof(delegateType));
+			}
+
+			AddMixinImpl(delegateType);
+		}
+
+		/// <summary>
+		///   Adds a delegate to be mixed into generated proxies. The <paramref name="delegate"/>
+		///   will act as the target for calls to a `Invoke` method with a signature matching that
+		///   of the delegate.
+		/// </summary>
+		/// <param name="delegate">The delegate that should act as the target for calls to `Invoke` methods with a matching signature.</param>
+		/// <exception cref="ArgumentNullException"><paramref name="delegate"/> is <see langword="null"/>.</exception>
+		public void AddDelegateMixin(MulticastDelegate @delegate)
```

Comment:
@jonorossi, what do you think, should the parameter type be `MulticastDelegate`, or `Delegate`? I'm using the one here, and the other over in `ProxyUtil`.

Reply:
Done. I've chosen to use `Delegate` in the public method signatures, but `MulticastDelegate` gets used for delegate type checks. See the last commit's message for details.